xinetd/service.c

```diff
@@ -249,6 +249,10 @@ static status_e activate_normal( struct service *sp )
    int                     v6on           = 0;
 #endif
 
+   if ( SC_BIND_DEVICE( scp ) )
+      if( setsockopt(SVC_FD(sp), SOL_SOCKET, SO_BINDTODEVICE, scp->sc_bind_device, strlen( scp->sc_bind_device ) ) )
```

Comment:
What happens on systems without the SO_BINDTODEVICE socket option?
Could this be conditionalized on whether SO_BINDTODEVICE is defined?
On systems that do not have this option, xinetd should probably refuse to enable the service and have a warning logged notifying the user that the configuration directive doesn't work on this system and that the service will be disabled until someone tells us what to do.